components/support/rate-limiter/src/lib.rs

```diff
@@ -0,0 +1,76 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Simple token bucket rate-limiter implementation.
+
+#[derive(Clone, Copy)]
+pub struct RateLimiter {
+    capacity: u8,
+    tokens: u8,
+    renewal_rate: f32, // per ms.
+    last_refill: u64,  // in ms.
+}
+
+impl RateLimiter {
+    pub fn new(capacity: u8, renewal_rate: f32) -> Self {
+        Self {
+            capacity,
+            tokens: capacity,
+            renewal_rate,
+            last_refill: Self::now(),
+        }
+    }
+
+    pub fn check(&mut self) -> bool {
+        self.refill();
+        if self.tokens == 0 {
+            return false;
+        }
+        self.tokens -= 1;
+        true
+    }
+
+    fn refill(&mut self) {
+        let now = Self::now();
+        let new_tokens = ((now - self.last_refill) as f64 * self.renewal_rate as f64) as u8; // `as` is a truncating/saturing cast.
+        if new_tokens > 0 {
+            self.last_refill = now;
+            self.tokens = std::cmp::min(self.capacity, self.tokens.saturating_add(new_tokens));
+        }
+    }
+
+    #[cfg(not(test))]
+    #[inline]
+    fn now() -> u64 {
+        // util::now()
```

Comment:
Just putting an explicit note on this commented-out bit so we don't forget about it.